### DISCUSSION TITLE
Grid and AC Loads Meters Showing Wrong Readings

### DISCUSSION THREAD

**Discussion by curaesolar** [mr-manuel/venus-os_dbus-enphase-envoy#14] [Q&A] [answered]
Hi,

@mr-manuel and thanks for this awesome integration. 

I have the same issue as @Manuel-Calero where the AC loads and Grid values are wrong.
[https://github.com/mr-manuel/venus-os_dbus-enphase-envoy/discussions/9#discussioncomment-7979893](url)

In my setup the Enphase system is connected to the AC input and in my config.ini I have set position to `0 = AC input 1`. 

```ini
[PV]
; Sum of the rated power (W) from all Microinverters connected to this Envoy
max = 4026
; position (connected input/output) of the PV inverter
; 0 = AC input 1 (select this if you don't have a MultiPlus or Quattro)
; 1 = AC output
; 2 = AC input 2
position = 0
```

I want to use the Multi and Battery as storage for excess solar and avoid using Grid power.

![image](https://github.com/user-attachments/assets/64da48db-4f71-4355-9f99-6569b1da8947)

Any guidance would be greatly appreciated.

Thanks.

**Comment by mr-manuel** [comment]
1. Please post a screenshot of your Enphase readings and Victron readings (Overview page, where you sell all the values).
2. Are you sure, that your grid readings are wrong? From your drawing you get the grid readings with an external current sensor of the MultiPlus? Did you configure that with the VE.Config software?
3. Is the PV reading correct?
4. What have you selected in the remote console/GUI under Menu -> Settings -> ESS -> Grid metering?

**Reply by curaesolar** [reply]
Hi @mr-manuel 

1. Here are the screenshots:
![enphase](https://github.com/user-attachments/assets/6155a90c-a480-477f-ba6b-f5b9d2834510)
![vrm](https://github.com/user-attachments/assets/8d0af8cc-7d36-4848-8f92-aafa2ae94e4a)

2. The grid readings don't match with the Enphase readings. The Enphase readings look correct to me.
3. Yes, the PV reading is correct. VE Configure 3 has External current sensor selected.
![image](https://github.com/user-attachments/assets/7bde0fa0-0df8-48e0-8e2c-80809be37a9e)

5. In the remote console/GUI under Menu -> Settings -> ESS -> Grid metering I have selected Inverter/Charger.
![image](https://github.com/user-attachments/assets/da104c6d-b32f-4034-b393-202fa031beed)

Thanks!

**Reply by curaesolar** [reply]
An example of the issue is now. The export to the Grid is incorrect and the AC load is also showing an incorrect value of 0.

![image](https://github.com/user-attachments/assets/d958019d-34e2-420e-82b1-8a2348cff4c7)

**Reply by mr-manuel** [reply]
Can you also post a screenshot of your connected devices?

**Reply by curaesolar** [reply]
Did you mean like this:
![image](https://github.com/user-attachments/assets/eb332be7-9884-49b1-8783-2b63be216d0c)

**Reply by mr-manuel** [reply]
Yes. So you are using only the Enpase PV input, which is correct, I assume. If yes, then you have a common system misconfiguration and your question fit better to the Victron Community.

**Reply by curaesolar** [reply]
@mr-manuel thanks for the guidance. I will check with the Victron community and feedback here to help anyone with the same problem.

**Comment by curaesolar** [comment, marked as answer]
Grid Meter is not optional in this setup as described [here](https://www.victronenergy.com/media/pg/Energy_Storage_System/en/ess-introduction---features.html).

An Energy Meter can be installed in the main distribution panel between the grid and the installation for a full or partial grid-parallel installation.

If there is any AC renewable energy source or any AC load between the grid connection point and the input side of the Multi/Quattro system, incorrect results will be calculated and recorded by the GX unless a grid meter is installed and enabled.

When renewable energy is being provided on the input side without a grid meter, **the grid value will be wrong (too low/negative)**; and

**The AC Load value shown will be too low** (and will show zero where there is a surplus of renewable energy).

Both issues are resolved by installing a grid meter.
